main.py

```diff
@@ -74,17 +81,24 @@ def colour_filter(col):
 switchover = 0
 
 if __name__=="__main__":
-    l.animate()
-    l.event_handler = event_handler
-    l.colour_filter = colour_filter
+    try:
+        l.animate()
+        l.event_handler = event_handler
+        l.colour_filter = colour_filter
+
+        sir.start()
+        server.start()
 
-    while 1:
-        print next_target
-        cur_thread = Thread(target=next_target, args=args)
+        while 1:
+            print next_target
+            cur_thread = Thread(target=next_target, args=args)
 
-        next_target = clock #if shit crashes
-        args = (l, )
+            next_target = clock #if shit crashes
+            args = (l, )
 
-        l.reset_state()
-        cur_thread.start()
-        cur_thread.join()
+            l.reset_state()
+            cur_thread.start()
+            cur_thread.join()
+    finally:
```

Comment:
this should be done with a `with` statement, instead of a `try .. finally` block.